src/fund/components/FundButton.tsx

```diff
@@ -19,34 +21,65 @@ export function FundButton({
   rel,
   target,
   text: buttonText = 'Fund',
+  successText: buttonSuccessText = 'Success',
+  errorText: buttonErrorText = 'Something went wrong',
+  state: buttonState = 'default',
+  onPopupClose,
+  onClick,
 }: FundButtonReact) {
   const componentTheme = useTheme();
   // If the fundingUrl prop is undefined, fallback to our recommended funding URL based on the wallet type
-  const fundingUrlToRender = fundingUrl ?? useGetFundingUrl();
+  const fallbackFundingUrl = useGetFundingUrl();
+  const fundingUrlToRender = fundingUrl ?? fallbackFundingUrl;
   const isDisabled = disabled || !fundingUrlToRender;
 
   const handleClick = useCallback(
     (e: React.MouseEvent) => {
       e.preventDefault();
+
       if (fundingUrlToRender) {
+        onClick?.();
         const { height, width } = getFundingPopupSize(
           popupSize,
           fundingUrlToRender,
         );
-        openPopup({
+        const popupWindow = openPopup({
           url: fundingUrlToRender,
           height,
           width,
           target,
         });
+
+        if (!popupWindow) {
+          return null;
+        }
+
+        const interval = setInterval(() => {
+          if (popupWindow?.closed) {
+            clearInterval(interval);
+            onPopupClose?.();
+          }
+        }, 500);
       }
     },
-    [fundingUrlToRender, popupSize, target],
+    [fundingUrlToRender, popupSize, target, onPopupClose, onClick],
   );
 
+  const buttonColorClass = useMemo(() => {
+    switch (buttonState) {
+      case 'error':
+        return pressable.error;
+      case 'loading':
+      case 'success':
+        return pressable.primary;
+      default:
+        return pressable.primary;
+    }
+  }, [buttonState]);
+
   const classNames = cn(
     componentTheme,
-    pressable.primary,
+    buttonColorClass,
     'px-4 py-3 inline-flex items-center justify-center space-x-2 disabled',
```

Comment:
Probably want to add hover:cursor-disabled here

Reply:
Sorry, meant cursor-not-allowed https://tailwindcss.com/docs/cursor#setting-the-cursor-style